CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+# CMakeLists.txt
+
+# Part of MR Lab Control Software
+
+# SPDX-License-Identifier: EUPL-1.2
+# SPDX-FileCopyrightText: Copyright (c) 2025 Martin Rumori/sonible GmbH
+
+cmake_minimum_required (VERSION 3.16)
+
+# This setting has to occur BEFORE the project call to work correctly
+set (CMAKE_OSX_DEPLOYMENT_TARGET "10.14" CACHE STRING "Minimum OS X deployment version" FORCE)
+
+project (mrlab VERSION 0.0.1)
+
+# CPM dependency management: https://github.com/cpm-cmake/CPM.cmake
+include (cmake/CPM.cmake)
+
+# Compile commands, useful for some IDEs like VS-Code
+set (CMAKE_EXPORT_COMPILE_COMMANDS TRUE)
+
+# Static linking in Windows
+set (CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+
+# JUCE dependency
+CPMAddPackage ("gh:juce-framework/JUCE#8.0.9")
+
+#====================================================================================
+juce_add_gui_app (mrlabctrl
+    PRODUCT_NAME  mrlabctrl
+    COMPANY_NAME  sonible
+    BUNDLE_ID     com.sonible.mrlabctrl
+
+    HARDENED_RUNTIME_ENABLED TRUE
+)
+
+set_target_properties (mrlabctrl
+    PROPERTIES
+    CXX_STANDARD 20
+)
+
+add_subdirectory (resources)
+add_subdirectory (src)
+
+target_include_directories (mrlabctrl PUBLIC "src")
+
+target_compile_definitions (mrlabctrl
+    PRIVATE
+        JUCE_WEB_BROWSER=0
+        JUCE_USE_CURL=0
+        JUCE_APPLICATION_NAME_STRING="MR Lab Control"
+        JUCE_APPLICATION_VERSION_STRING="$<TARGET_PROPERTY:mrlabctrl,JUCE_VERSION>"
```

Comment:
Shouldn't these JUCE specific defines be settable through the `juce_add_gui_app` call somehow?

Reply:
- [ ] check a better way to propagate version and app name